File: backend/app/schemas/statistics.py
from pydantic import BaseModel
from typing import Dict, List, Optional


class OverviewStats(BaseModel):
    """总览统计"""
    # 粮食统计
    total_crops: int
    growing_crops: int
    harvested_crops: int
    total_crop_yield: float

    # 动物统计
    total_animal_varieties: int
    total_animals: int
    estimated_daily_yield: float

    # 花卉统计
    total_flower_varieties: int
    total_flowers: int


class CropStats(BaseModel):
    """粮食统计"""
    by_variety: Dict[str, float]  # 品种 -> 产量
    by_status: Dict[str, int]  # 状态 -> 数量


class AnimalStats(BaseModel):
    """动物统计"""
    by_product_type: Dict[str, int]  # 产品类型 -> 数量
    by_variety: Dict[str, int]  # 品种 -> 数量


class FlowerStats(BaseModel):
    """花卉统计"""
    by_season: Dict[str, int]  # 季节 -> 数量
    by_purpose: Dict[str, int]  # 用途 -> 数量


class ChartData(BaseModel):
    """图表数据格式"""
    title: str
    type: str  # bar, line, pie, etc.
    labels: List[str]
    datasets: List[Dict]


class ChartDataResponse(BaseModel):
    """图表数据汇总响应"""
    crop_yield_by_variety: ChartData
    crop_status_pie: ChartData
    animal_by_product: ChartData
    flower_by_season: ChartData


class CalendarEvent(BaseModel):
    """日历事件"""
    date: str
    title: str
    type: str  # plant, harvest, bloom
    category: str  # crop, animal, flower
    count: int


class CalendarData(BaseModel):
    """日历数据响应"""
    events: List[CalendarEvent]

File: frontend/src/types/statistics.ts
// 统计数据类型定义
export interface OverviewStats {
  total_crops: number
  growing_crops: number
  harvested_crops: number
  total_crop_yield: number
  total_animal_varieties: number
  total_animals: number
  estimated_daily_yield: number
  total_flower_varieties: number
  total_flowers: number
}

export interface CropStats {
  by_variety: Record<string, number>
  by_status: Record<string, number>
}

export interface AnimalStats {
  by_product_type: Record<string, number>
  by_variety: Record<string, number>
}

export interface FlowerStats {
  by_season: Record<string, number>
  by_purpose: Record<string, number>
}

export interface ChartData {
  title: string
  type: string
  labels: string[]
  datasets: Array<Record<string, unknown>>
}

export interface ChartDataResponse {
  crop_yield_by_variety: ChartData
  crop_status_pie: ChartData
  animal_by_product: ChartData
  flower_by_season: ChartData
}

export interface CalendarEvent {
  date: string
  title: string
  type: string
  category: string
  count: number
}

export interface CalendarData {
  events: CalendarEvent[]
}
